keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -245,28 +245,52 @@ class TitusClusterHandler(
       user = DEFAULT_SERVICE_ACCOUNT
     )
 
-    val image = images.find { it.digest == container.digest }
-      ?: throw ExportError("Unable to find matching image (searching by digest) in registry ($registry) for $container")
-    val tagVersionStrategy = findTagVersioningStrategy(image)
-      ?: throw DockerArtifactExportError(image.toString(), container.toString())
+    val matchingImages = images.filter { it.digest == container.digest }
+    if (matchingImages.isEmpty()) {
+      throw ExportError("Unable to find matching image (searching by digest) in registry ($registry) for $container")
+    }
+    val versionStrategy = guessVersioningStrategy(matchingImages)
+      ?: throw DockerArtifactExportError(matchingImages.map { it.tag }, container.toString())
 
     return DockerArtifact(
       name = container.repository(),
-      tagVersionStrategy = tagVersionStrategy
+      tagVersionStrategy = versionStrategy
     )
   }
 
-  fun findTagVersioningStrategy(image: DockerImage): TagVersionStrategy? {
-    if (image.tag.toIntOrNull() != null) {
+  /**
+   * Tries to find a matching versioning strategy from a list of docker tags that correspond to the same digest.
+   */
+  fun guessVersioningStrategy(images: List<DockerImage>): TagVersionStrategy? {
+    val versioningStrategies = mutableSetOf<TagVersionStrategy>()
+
+    images.forEach { image ->
+      val versionStrategy = calculateVersioningStrategyFrom(image.tag)
+      if (versionStrategy != null) {
+        versioningStrategies.add(versionStrategy)
+      }
+    }
+    return when (versioningStrategies.size) {
+      1 -> versioningStrategies.first()
+      0 -> null
+      else -> {
+        log.warn("Multiple versioning strategies apply for image, returning first")
+        versioningStrategies.first()
+      }
+    }
+  }
+
+  fun calculateVersioningStrategyFrom(tag: String): TagVersionStrategy? {
```

Comment:
Not sure this is the name @gal-yardeni found confusing, but maybe `extractVersioningStrategy` or `deriveVersioningStrategy`?

Reply:
I like `deriveVersioningStrategy` better, thanks for the suggestion